pkg/external/machineactuator.go

```diff
@@ -160,34 +161,45 @@ func (c *ExternalClient) executeAction(command string, cluster *clusterv1.Cluste
 	})
 
 	if err != nil {
-		return nil, err
+		return false, err
 	}
 
 	if doWait {
-		if err := c.waitForJob(fencingJob.Name, fencingJob.Namespace, -1); err != nil {
+		nodeFenceState, err := c.waitForJob(fencingJob.Name, fencingJob.Namespace, -1)
+		if err != nil {
 			glog.Errorf("Job %v error: %v", fencingJob.Name, err)
-			return nil, err
+			return false, err
 		}
+		return nodeFenceState, nil
 	}
 
 	glog.Infof("Job %v complete", fencingJob.Name)
-	return &fencingJob.Name, nil
+	return true, nil
 }
 
-func (c *ExternalClient) waitForJob(jobName string, namespace string, retries int) error {
+func (c *ExternalClient) waitForJob(jobName string, namespace string, retries int) (bool, error) {
 	// TODO: Use informers to fetch resources
 	job, err := c.kubeclient.BatchV1().Jobs(namespace).Get(jobName, metav1.GetOptions{})
-	glog.Infof("Waiting %d times for job %v: %v", retries, job.Name, err)
+	if err != nil {
+		return false, err
+	}
+	glog.Infof("Waiting %d times for job %v", retries, job.Name)
 
+	var jobPod *corev1.Pod
 	for lpc := 0; lpc < retries || retries < 0; lpc++ {
-		if err != nil {
-			return err
+		if jobPod != nil {
+			for _, containerState := range jobPod.Status.ContainerStatuses {
+				if containerState.State.Terminated != nil {
+					if containerState.State.Terminated.ExitCode == 2 {
+						return false, nil
```

Comment:
What if the job succeeds when it runs again but you've already reported false to the caller?